insync: 3.3.5.40925 -> 3.8.5.50499

hellwolf opened the pull request:
## Description of changes

The `insync-v3` package is currently broken in both `master` and `22.11`.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Resolution

- Deleting insync-v3 from top-level, superseded by insync.
- Using buildFHSEnv to simplify packaging process.
- Drop backport attempt for 22.11 for this package going forward.

### Related Issues

- [BLOCKED] #207829 - Because it has dependencies on python 3.7 packages pyside2 and shiboken2, it also depends on pull request.
- [RELATED] closes #186805
- [RELATED] closes #139728

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

hellwolf commented:
Current the binary crashes, will find the root cause of it.

hellwolf commented:
Update: I managed to make it work, but I still need to make shiboken2 and pyside2 to compile for python 2.7, I am merging upstream code then pushing to this pull request soon.

![image](https://user-images.githubusercontent.com/186660/209491547-7f34fb1b-0a16-4790-bcc1-0ecf04caea22.png)

hellwolf commented:
Pulled out the python package pull requests to 207829

benley commented:
When you are eventually ready to merge this, would you like to take over maintainership of the derivation in nixpkgs?  I'm going to remove myself from the meta.maintainers line since I haven't been using insync for a few years now.

hellwolf commented:
> When you are eventually ready to merge this, would you like to take over maintainership of the derivation in nixpkgs? I'm going to remove myself from the meta.maintainers line since I haven't been using insync for a few years now.

Happy to do so, since I am still using it as part of the workflow. I don't know how to add myself to maintainers list yet, I will look into it.

hellwolf commented:
For some reason, the system tray icon is not visible. I am using `gnomeExtensions.appindicator`.

It seems a chronic bug in upstream insync, I will consider it a known bug and mark this PR ready for now.

hellwolf commented:
@benley do you want me to update the maintainer-list.nix and change the meta.maintainers in the same pull request.

benley commented:
doing some testing now!

benley commented:
I am still having trouble getting insync to start with this PR, both as-is and rebased onto nixpkgs master.  Might I be missing something?

```
$ nix-build -A insync-v3
... 15 minutes of building pyside2 later ...
$ result/bin/insync start --no-daemon
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
/nix/store/xch0pgrcpmzs4pvx2iabd603hg8xmpp8-insync-3.8.4.50481/lib/insync/insync: symbol lookup error: /nix/store/xch0pgrcpmzs4pvx2iabd603hg8xmpp8-insync-3.8.4.50481/lib/insync/PySide2/plugins/platforminputcontexts/libibusplatforminputcontextplugin.so: undefined symbol: _ZdlPvm, version Qt_5
```

hellwolf commented:
> I am still having trouble getting insync to start with this PR, both as-is and rebased onto nixpkgs master. Might I be missing something?
> 
> ```
> $ nix-build -A insync-v3
> ... 15 minutes of building pyside2 later ...
> $ result/bin/insync start --no-daemon
> Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
> /nix/store/xch0pgrcpmzs4pvx2iabd603hg8xmpp8-insync-3.8.4.50481/lib/insync/insync: symbol lookup error: /nix/store/xch0pgrcpmzs4pvx2iabd603hg8xmpp8-insync-3.8.4.50481/lib/insync/PySide2/plugins/platforminputcontexts/libibusplatforminputcontextplugin.so: undefined symbol: _ZdlPvm, version Qt_5
> ```

That's interesting, which nixpkgs channel are you on?

benley commented:
> That's interesting, which nixpkgs channel are you on?
I'm running nixos-unstable (probably a week or two out of date), but I built and ran insync-v3 from your PR branch, and also tried rebasing it onto master.

hellwolf commented on pkgs/applications/networking/insync/v3.nix:
```diff
@@ -3,47 +3,56 @@
-    wayland
-  ];
+    qtserialport
+  ]) ++ (with python37Packages; [
```
Unfortunately insync's pyside plugins are compiled with them:

```
$ find result/lib/insync -name '*37m*' 
result/lib/insync/_struct/cpython-37m-x86_64-linux-gnu
result/lib/insync/_struct/cpython-37m-x86_64-linux-gnu/sotruct.cpython-37m-x86_64-linux-gnu.so
result/lib/insync/lib-dynload/array.cpython-37m-x86_64-linux-gnu.so
...
```

Could try a different binary some day: https://www.insynchq.com/downloads/linux#ubuntu, but unfortunately currently ubuntu kinetic build is using glibc 2.36

SuperSandro2000 replied:
We should definitely try that. Python 3.7 is EOL and it will probably fall apart eventually.

hellwolf commented:
> > That's interesting, which nixpkgs channel are you on?
> 
> I'm running nixos-unstable (probably a week or two out of date), but I built and ran insync-v3 from your PR branch, and also tried rebasing it onto master.
> 
> (edit: formatting oops)

that's a pitty... 

I found a similar issue in https://github.com/NixOS/nixpkgs/pull/194779

I will look into more.

hellwolf commented:
> > That's interesting, which nixpkgs channel are you on?
> 
> I'm running nixos-unstable (probably a week or two out of date), but I built and ran insync-v3 from your PR branch, and also tried rebasing it onto master.
> 
> (edit: formatting oops)

fwiw, mine looks like:

```
$ ldd result/lib/insync/PySide2/plugins/platforminputcontexts/libibusplatforminputcontextplugin.so 
ldd: warning: you do not have execution permission for `result/lib/insync/PySide2/plugins/platforminputcontexts/libibusplatforminputcontextplugin.so'
	linux-vdso.so.1 (0x00007fff22bfd000)
	libQt5Gui.so.5 => /nix/store/9lcgwnc70f4wj1czklczql7akwcv24mi-qtbase-5.15.7/lib/libQt5Gui.so.5 (0x00007ff2ffd90000)
	libQt5DBus.so.5 => /nix/store/9lcgwnc70f4wj1czklczql7akwcv24mi-qtbase-5.15.7/lib/libQt5DBus.so.5 (0x00007ff30067d000)
	libQt5Core.so.5 => /nix/store/9lcgwnc70f4wj1czklczql7akwcv24mi-qtbase-5.15.7/lib/libQt5Core.so.5 (0x00007ff2ff7e0000)
	libGL.so.1 => /nix/store/ykjcsxdh9c1w664g6v38d86gph8m6mq7-libglvnd-1.5.0/lib/libGL.so.1 (0x00007ff2ff752000)
	libpthread.so.0 => /nix/store/4nlgxhb09sdr51nc9hdm8az5b08vzkgx-glibc-2.35-163/lib/libpthread.so.0 (0x00007ff300678000)
	libstdc++.so.6 => /nix/store/wprxx5zkkk13hpj6k1v6qadjylh3vq9m-gcc-11.3.0-lib/lib/libstdc++.so.6 (0x00007ff2ff53c000)
	libm.so.6 => /nix/store/4nlgxhb09sdr51nc9hdm8az5b08vzkgx-glibc-2.35-163/lib/libm.so.6 (0x00007ff2ff45c000)
	libgcc_s.so.1 => /nix/store/wprxx5zkkk13hpj6k1v6qadjylh3vq9m-gcc-11.3.0-lib/lib/libgcc_s.so.1 (0x00007ff30065c000)
	libc.so.6 => /nix/store/4nlgxhb09sdr51nc9hdm8az5b08vzkgx-glibc-2.35-163/lib/libc.so.6 (0x00007ff2ff253000)
	libpng16.so.16 => /nix/store/8gmgb7m65nafhhqksc71bsfykggsx2ps-libpng-apng-1.6.37/lib/libpng16.so.16 (0x00007ff2ff219000)
	libz.so.1 => /nix/store/zaflwh2nwzj1f0wngd7hqm3nvlf3yhsx-zlib-1.2.13/lib/libz.so.1 (0x00007ff30063c000)
	libharfbuzz.so.0 => /nix/store/9q6s2jbnaikjj7lhxpgy5zaz9biw68bz-harfbuzz-5.3.1/lib/libharfbuzz.so.0 (0x00007ff2ff111000)
	libdbus-1.so.3 => /nix/store/lyvgw6wxb61l2c1krpr2hzprd34cms4v-dbus-1.14.4-lib/lib/libdbus-1.so.3 (0x00007ff2ff0b9000)
	libicui18n.so.72 => /nix/store/wkmc51s1nyyic7ym9c44k57wk8aqlpp9-icu4c-72.1/lib/libicui18n.so.72 (0x00007ff2fed7a000)
	libicuuc.so.72 => /nix/store/wkmc51s1nyyic7ym9c44k57wk8aqlpp9-icu4c-72.1/lib/libicuuc.so.72 (0x00007ff2feb75000)
	libicudata.so.72 => /nix/store/wkmc51s1nyyic7ym9c44k57wk8aqlpp9-icu4c-72.1/lib/libicudata.so.72 (0x00007ff2fcda5000)
	libpcre2-16.so.0 => /nix/store/b33ikpmz4b1gd654v5s62phykmpd58fp-pcre2-10.40/lib/libpcre2-16.so.0 (0x00007ff2fcd15000)
	libzstd.so.1 => /nix/store/9vgv9wx5wf0311j8mxb8zkzc873mnicb-zstd-1.5.2/lib/libzstd.so.1 (0x00007ff2fcc51000)
	libgthread-2.0.so.0 => /nix/store/61qy7d70070djqcg135l22v18a5py758-glib-2.74.3/lib/libgthread-2.0.so.0 (0x00007ff300635000)
	libglib-2.0.so.0 => /nix/store/61qy7d70070djqcg135l22v18a5py758-glib-2.74.3/lib/libglib-2.0.so.0 (0x00007ff2fcb14000)
	/nix/store/4nlgxhb09sdr51nc9hdm8az5b08vzkgx-glibc-2.35-163/lib64/ld-linux-x86-64.so.2 (0x00007ff300711000)
	libGLX.so.0 => /nix/store/ykjcsxdh9c1w664g6v38d86gph8m6mq7-libglvnd-1.5.0/lib/libGLX.so.0 (0x00007ff2fcae0000)
	libX11.so.6 => /nix/store/3zk3527zglm18ykrvgjdhgz3a4zrfapr-libX11-1.8.1/lib/libX11.so.6 (0x00007ff2fc99b000)
	libXext.so.6 => /nix/store/z1zr4xyiaxvdkxc0f1a5cswkvgmix3dw-libXext-1.3.4/lib/libXext.so.6 (0x00007ff2fc986000)
	libGLdispatch.so.0 => /nix/store/ykjcsxdh9c1w664g6v38d86gph8m6mq7-libglvnd-1.5.0/lib/libGLdispatch.so.0 (0x00007ff2fc8ce000)
	libfreetype.so.6 => /nix/store/f1lgywhbb9frvkjkjkbgw2yp63hc462p-freetype-2.12.1/lib/libfreetype.so.6 (0x00007ff2fc806000)
	libgraphite2.so.3 => /nix/store/f0acn4l3g0yiavxjx4c68wad03ah67iq-graphite2-1.3.14/lib/libgraphite2.so.3 (0x00007ff2fc7de000)
	libsystemd.so.0 => /nix/store/iap1if4pm6h6sacydm344p38j182c189-systemd-minimal-252.1/lib/libsystemd.so.0 (0x00007ff2fc6f4000)
	libpcre2-8.so.0 => /nix/store/b33ikpmz4b1gd654v5s62phykmpd58fp-pcre2-10.40/lib/libpcre2-8.so.0 (0x00007ff2fc656000)
	libxcb.so.1 => /nix/store/6031yyvcgc64qm40s5vj7rblz7fwbvnj-libxcb-1.14/lib/libxcb.so.1 (0x00007ff2fc62b000)
	libbz2.so.1 => /nix/store/kxdyx0dqy8gpx8yp61wybn2dr1va0vn8-bzip2-1.0.8/lib/libbz2.so.1 (0x00007ff2fc618000)
	libcap.so.2 => /nix/store/9vfggw8qmvdxhb1yj9yf8crqndg5xc08-libcap-2.66-lib/lib/libcap.so.2 (0x00007ff300623000)
	libXau.so.6 => /nix/store/fh0ih93xv6ifvgidjknydvc950mlch3r-libXau-1.0.9/lib/libXau.so.6 (0x00007ff2fc611000)
	libXdmcp.so.6 => /nix/store/iwb5zaid6mxkivhl5p0xykgmsdj0mzjv-libXdmcp-1.1.3/lib/libXdmcp.so.6 (0x00007ff2fc609000)

$ nix-build $PWD -A insync-v3
/nix/store/xch0pgrcpmzs4pvx2iabd603hg8xmpp8-insync-3.8.4.50481
```

But I guess we should trust reproduciable builds...

hellwolf commented:
> I am still having trouble getting insync to start with this PR, both as-is and rebased onto nixpkgs master. Might I be missing something?
> 
> ```
> $ nix-build -A insync-v3
> ... 15 minutes of building pyside2 later ...
> $ result/bin/insync start --no-daemon
> Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
> /nix/store/xch0pgrcpmzs4pvx2iabd603hg8xmpp8-insync-3.8.4.50481/lib/insync/insync: symbol lookup error: /nix/store/xch0pgrcpmzs4pvx2iabd603hg8xmpp8-insync-3.8.4.50481/lib/insync/PySide2/plugins/platforminputcontexts/libibusplatforminputcontextplugin.so: undefined symbol: _ZdlPvm, version Qt_5
> ```

Actually I got that during exiting of the insync instead:

```
$ ./result/bin/insync start --no-daemon
QXcbIntegration: Cannot create platform OpenGL context, neither GLX nor EGL are enabled
QSystemTrayIcon::setVisible: No Icon set
INFO     2022-12-29 16:40:36,692 [mainlogs:_log_run:138] Core(app_version=3.8.4.50481, platform=Linux-x86_64-nixos/22.11) initialized
INFO     2022-12-29 16:40:36,739 [mainsettings:_load_settings:236] Loaded settings, machine fingerprint = b29xkLEt4XijophExAsrl-hTWys=
INFO     2022-12-29 16:40:36,839 [inotify_manager:__init__:132] INotifyManager FS_ENCODING: utf-8
INFO     2022-12-29 16:40:36,840 [fswatcher:_start:57] LinuxFSWatcher._start
INFO     2022-12-29 16:40:36,840 [inotify_manager:_pull_loop:315] Inotify loop enter
INFO     2022-12-29 16:40:37,032 [fswatcher:watch:77] watch origin: /home/hellwolf/Documents/Personal/Google
INFO     2022-12-29 16:40:37,035 [app:start_core:65] core started
INFO     2022-12-29 16:40:37,035 [unix_socket_server:start:106] unix socket server thread start
js: Access to XMLHttpRequest at 'https://app.posthog.com/decide/?v=2&ip=1&_=1672324837152&ver=1.26.0' from origin 'file://' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, chrome-untrusted, https.
js: Bad HTTP status: 0 
js: Enqueued failed request for retry in 6000
js: Access to XMLHttpRequest at 'https://app.posthog.com/e/?ip=1&_=1672324837336&ver=1.26.0' from origin 'file://' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, chrome-untrusted, https.
js: Bad HTTP status: 0 
js: Enqueued failed request for retry in 6000
js: Access to XMLHttpRequest at 'https://app.posthog.com/decide/?v=2&ip=1&_=1672324838052&ver=1.26.0' from origin 'file://' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, chrome-untrusted, https.
js: Bad HTTP status: 0 
js: Enqueued failed request for retry in 6000
js: Access to XMLHttpRequest at 'https://app.posthog.com/e/?ip=1&_=1672324841053&ver=1.26.0' from origin 'file://' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, chrome-untrusted, https.
js: Bad HTTP status: 0 
js: Enqueued failed request for retry in 6000
INFO     2022-12-29 16:40:44,629 [app:quit:80] quitting
INFO     2022-12-29 16:40:44,630 [unix_socket_server:stop:126] unix socket server thread stop
INFO     2022-12-29 16:40:44,636 [fswatcher:_stop:70] LinuxFSWatcher._stop
INFO     2022-12-29 16:40:44,659 [app:stop:75] core stopped
INFO     2022-12-29 16:40:44,660 [app:stop:77] clean exit
/nix/store/xch0pgrcpmzs4pvx2iabd603hg8xmpp8-insync-3.8.4.50481/lib/insync/insync: symbol lookup error: /nix/store/xch0pgrcpmzs4pvx2iabd603hg8xmpp8-insync-3.8.4.50481/lib/insync/PySide2/plugins/platforminputcontexts/libibusplatforminputcontextplugin.so: undefined symbol: _ZdlPvm, version Qt_5
```

hellwolf commented:
Here you go, I got lots of great help from @lilyinstarlight  in NixOS matrix dev channel. This new version should work more reliably now.

@benley please test again!

Stefanomarton commented:
Hey there !! 
I don't want to bother but what is the actual state of insync?
I tried to install it from the unstable channel but I encountered several problems. 

Thanks 
~S

Zahrun commented:
I added insync from this PR to my config and it works perfectly fine. The only issue I see is that the tray icon has no image.
![image](https://user-images.githubusercontent.com/10415894/215740572-9276bdc6-9bd0-4c3b-92db-b0cbc2b22008.png)

hellwolf commented:
Hey all, yes I have been using, happy to push to merge if more people tested it.

And yes, a known issue is that the insync tray icon doesn't show image correctly. I have fought this issue with  @lilyinstarlight for a few hours, and the conclusion was it was it's very likely due to insync packaging decided to sanitizing the input required XDG_ environment variable and causing not reading icons correctly on Nix.

If you all okay with this known bug before we can find a new workaround, I will push to the right force to help us merge this!

Artturin reviewed (changes requested):
fixup the commits

hellwolf commented on maintainers/maintainer-list.nix:
```diff
@@ -5586,6 +5586,12 @@
     githubId = 86223025;
     name = "helium";
   };
+  hellwolf = {
```
done!

Artturin replied:
the maintainers commit should be before `insync-v3: fix build`

hellwolf replied:
np, done

Artturin commented:
just noticed that python37 has been removed from nixpkgs

hellwolf commented:
Silly me, I synced with master branch today, and I didn't realize python37 was removed. fwiw, you can apply this patch to 22.11 branch if you need it urgently.

Unfortunately insync has silly packaging, currently this one relies on python 3.7. The other package from insync against the latest ubuntu though relies on glibc 2.36, so it can't be used neither... I will watch the situation and see what I can do to fix it..

lilyinstarlight commented:
For issue/pr cross-referencing, the glibc 2.36 update is #188492

hellwolf commented:
Taking a note re libpython3 libpython3.x. https://stackoverflow.com/questions/39863827/what-is-libpython3-so-compared-with-libpython3-5m-so-built-from-python-3-5-2-sou for future reference and people who are also curious about it.

hellwolf commented:
@Artturin would it still make sense to back port it to 22.11 where python37 is available, while hope insync's new binary moving to newer python?

hellwolf commented:
Hey folks, would it be a terrible idea if we simply wrap this with `run-steam`?

SuperSandro2000 commented on pkgs/applications/networking/insync/v3.nix:
```diff
@@ -1,73 +1,115 @@
+    # fix the launcher
+    rm -f "$out/bin/insync"
+    makeWrapper $out/lib/insync/insync $out/bin/insync \
+      --prefix XDG_DATA_DIRS : $GSETTINGS_SCHEMAS_PATH \
+      --prefix XDG_DATA_DIRS : $out/share
```
```suggestion
      --prefix XDG_DATA_DIRS : $GSETTINGS_SCHEMAS_PATH:$out/share
```

SuperSandro2000 commented on pkgs/applications/networking/insync/v3.nix:
```diff
@@ -1,73 +1,115 @@
+    ln -s ${python37}/lib/libpython3.7m.so $out/lib/insync/libpython3.7m.so.1.0
+    ln -s ${python37Packages.shiboken2}/lib/python3.7/site-packages/shiboken2 $out/lib/insync/shiboken2
+    mkdir $out/lib/insync/PySide2/
+    ln -s ${python37Packages.pyside2}/lib/python3.7/site-packages/PySide2/* $out/lib/insync/PySide2/
```
```suggestion
    ln -s ${python37Packages.pyside2}/${python37.sitePakcages}/PySide2/* $out/lib/insync/PySide2/
```

hellwolf replied:
no longer in the packaging process.

hellwolf commented:
Okay, actually it may be a good time to ask everyone in this thread a question, considering where we are:

1. Upstream insync-v3 does not seem to move on from python37 anytime soon.
2. As a result, packaging is quite a challenge since all python37 dependencies must be rebuilt for this package, especially pyside for Qt.
3. Currently only on 22.11 this package can still be built, though some fundamental issue still persists re notification try icon.

Instead of giving up and abandoning this package, I have a proposal, how about we change it to using steam-run as runner instead? Downside is that whoever needs this package is damned to have the entire steam-run dependency; but upside is it becomes much easier to package.

I will give two vote options, please signal your preference:

1. :+1: yes steam-run
2. :-1:  noooooo

hellwolf commented:
Good news folks! It works now! Without using full steam-run, instead with a very minimum FHS! Please give it a try!

Although the system tray icon still not working:

```
$ result/bin/insync-3.8.5.50499 start --no-daemon 
QApplication: invalid style override passed, ignoring it.
    Available styles: Windows, Fusion
QSystemTrayIcon::setVisible: No Icon set
```

I will give a few more debugging of this before call it known bug and consider we are good to go for this package again!

hellwolf commented:
@ofborg build insync

hellwolf committed:
insync: superceded by insync-v3

hellwolf committed:
insync-v3: remove

insync-v3 has been merged into the insync package; use insync instead.

hellwolf committed:
insync: 3.3.5.40925 -> 3.8.5.50499

- Simplified packaging process:
  - Use as much vendor provided binaries as possible.
  - Only replace glibc with system provided ones (2.36+ required, hence
    nixos 22.11 is out).
  - Use buildFHSEnv to create FHS environment to run insync in.

Known bug(s):
  1) Currently the system try icon does not render correctly.
  2) libqtvirtualkeyboardplugin does not have necessary Qt library shipped from vendor.

hellwolf commented:
Everyone this is ready now!!

* Removed `insync-v3` from top-level, with only `insync`  superseding it.
* insync: 3.3.5.40925 -> 3.8.5.50499
    - Simplified packaging process:
      - Use as much vendor provided binaries as possible.
      - Only replace glibc with system provided ones (2.36+ required, hence
        nixos 22.11 is out).
      - Use buildFHSEnv to create FHS environment to run insync in.
    - Known bug(s):
      1) Currently the system try icon does not render correctly.
      2) libqtvirtualkeyboardplugin does not have necessary Qt library shipped from vendor.

SuperSandro2000 commented on pkgs/applications/networking/insync/default.nix:
```diff
@@ -38,8 +30,85 @@ stdenv.mkDerivation rec {
+    # NB! This did the trick, otherwise it segfaults! However I don't understand why!
+    dontStrip = true;
+  };
+in buildFHSEnv { # ref: pkgs/build-support/build-fhsenv-bubblewrap/default.nix
+  name = "${pname}-${version}";
+  inherit meta;
+
+  # for including insync's xdg data dirs
+  extraOutputsToInstall = [ "share" ];
+
+  targetPkgs = pkgs: [
+    insync-pkg
+  ];
+
+  multiPkgs = pkgs: with pkgs; [
+    # apparently only package needed for the FHS :)
+    libudev0-shim
+  ];
+
+  runScript = writeShellScript "insync-wrapper.sh" ''
+    export QT_STYLE_OVERRIDE=Fusion
+    exec "${insync-pkg.outPath}/lib/insync/insync" "$@"
+  '';
+  # "insync start" command starts a daemon.
+  dieWithParent = false;
```
```suggestion
in buildFHSEnv {
  name = "${pname}-${version}";
  inherit pname version;
  
  # for including insync's xdg data dirs
  extraOutputsToInstall = [ "share" ];

  targetPkgs = pkgs: [
    insync-pkg
  ];

  multiPkgs = pkgs: with pkgs; [
    libudev0-shim
  ];

  runScript = writeShellScript "insync-wrapper.sh" ''
    export QT_STYLE_OVERRIDE=Fusion
    exec "${insync-pkg.outPath}/lib/insync/insync" "$@"
  '';

  # "insync start" command starts a daemon.
  dieWithParent = false;
  
  inherit meta;
```
Do we really must enforce the qt style?
Do we need to use multiPkgs aka 32 and 64 bit?

hellwolf replied:
> Do we really must enforce the qt style?

Not necessarily, it was added to suppress one warning from Qt output.

>  Do we need to use multiPkgs aka 32 and 64 bit?

No, only 64bit binaries are shipped from the vendor.

SuperSandro2000 commented on pkgs/applications/networking/insync/default.nix:
```diff
@@ -38,8 +30,85 @@ stdenv.mkDerivation rec {
-    broken = true;
   };
+  insync-pkg = stdenv.mkDerivation {
+    inherit pname;
+    inherit version;
+    inherit meta;
```
```suggestion
    inherit pname version;
```

SuperSandro2000 commented on pkgs/applications/networking/insync/default.nix:
```diff
@@ -38,8 +30,85 @@ stdenv.mkDerivation rec {
+    dontBuild = true;
+
+    # NB! This did the trick, otherwise it segfaults! However I don't understand why!
+    dontStrip = true;
```
```suggestion
    dontStrip = true;

    inherit meta;
```

SuperSandro2000 commented on pkgs/applications/networking/insync/default.nix:
```diff
@@ -38,8 +30,85 @@ stdenv.mkDerivation rec {
+      cp -R usr/* $out/
+
+      # use system glibc
+      rm -f $out/lib/insync/libgcc_s.so.1
+      rm -f $out/lib/insync/libstdc++.so.6
```
```suggestion
      rm -f $out/lib/insync/{libgcc_s.so.1,libstdc++.so.6}
```

SuperSandro2000 commented on pkgs/applications/networking/insync/default.nix:
```diff
@@ -38,8 +30,85 @@ stdenv.mkDerivation rec {
+      rm -f $out/lib/insync/libstdc++.so.6
+
+      # remove badly packaged plugins
+      rm -f $out/lib/insync/PySide2/plugins/platforminputcontexts/libqtvirtualkeyboardplugin.so
```
```suggestion
      rm $out/lib/insync/PySide2/plugins/platforminputcontexts/libqtvirtualkeyboardplugin.so
```

SuperSandro2000 commented on pkgs/applications/networking/insync/default.nix:
```diff
@@ -38,8 +30,85 @@ stdenv.mkDerivation rec {
+      dpkg-deb --fsys-tarfile $src | tar -x --no-same-permissions --no-same-owner
+    '';
+
+    preBuild = ''
+      addAutoPatchelfSearchPath $out/lib/insync/
+    '';
```
Do we need this even when using fhs?

hellwolf replied:
yep, no longer needed, eliminated.

hellwolf commented:
> Please don't use steam-run as it is giant and has a proprietary dependencies on steam, use a plain fhs instead and add the libraries that are required to it.

Yep, using `buildFHSEnv` now after other recommending, works like a charm.

hellwolf commented:
Anything more I can do to help to get this merged?

benley reviewed (approved):
Let's give @SuperSandro2000 a chance to chime in once more before merging but this looks pretty solid to me!

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/2199

SuperSandro2000 commented on pkgs/top-level/all-packages.nix:
```diff
@@ -33912,8 +33912,6 @@ with pkgs;
 
   insync = callPackage ../applications/networking/insync { };
 
-  insync-v3 = libsForQt5.callPackage ../applications/networking/insync/v3.nix { };
```
we should add an alias for this

hellwolf replied:
Is it still good to keep this -v3 alias though? maybe time to deprecate it if there is no intention to keep maintaining multiple versions going forward?

SuperSandro2000 replied:
see https://github.com/NixOS/nixpkgs/blob/master/doc/contributing/submitting-changes.chapter.md#:~:text=Add%20an%20alias%20for%20the%20package%20name%20in%20pkgs/top%2Dlevel/aliases.nix and https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/aliases.nix#L3-L18

hellwolf replied:
Thank you for pointing this out, I amended the commits!

hellwolf commented:
Good day everyone! I have fixed the alias issue as suggested by @SuperSandro2000 , can this still make to the [2023-05-15 timeline](https://github.com/NixOS/nixpkgs/issues/223562)?

SuperSandro2000 commented:
That timeline is not relevant for unimportant leave packages.
